### DISCUSSION TITLE
It seems that jlama does not support Chinese prompt, the content was truncated.

### DISCUSSION THREAD

**Discussion by zhouruicn** [tjake/Jlama#107] [Q&A]
I use the test case from "[jlama-tests](https://github.com/tjake/Jlama/tree/main/jlama-tests)"

 @Test
    @Order(1)
    public void LlamaRun() throws Exception {
        var dir = appContext.nodeContext().nodePath().commonsLlm();
        try (WeightLoader weights = SafeTensorSupport.loadWeights(dir.toFile())) {
            LlamaTokenizer tokenizer = new LlamaTokenizer(dir);
            Config c = om.readValue(dir.resolve("config.json").toFile(), LlamaConfig.class);
            LlamaModel model = new LlamaModel(c, weights, tokenizer, DType.F32, DType.F32,
                            Optional.empty());

            PromptSupport.Builder builder = model.promptSupport().get().builder();
            builder.addUserMessage("您能讲中文吗");// chinese text here! it means can you speek chinese?
            builder.addGenerationPrompt(true);
            PromptContext promptContext = builder.build();

            LOGGER.info("First prompt \n{}", promptContext);
            Generator.Response r =
                            model.generate(UUID.randomUUID(), promptContext, 0.7f, 1024, (l, f) -> {
                            });
            LOGGER.info("Response: {}", r.responseText);
        }
    }

3:50:04.938 [main] DEBUG com.github.tjake.jlama.safetensors.SafeTensorIndex -- Adding split 0-2008209408 with 451 tensors of 451
23:50:05.954 [main] INFO com.github.tjake.jlama.model.AbstractModel -- Model type = Q4, Working memory type = F32, Quantized memory type = F32
23:50:06.037 [main] INFO com.x.app.ai.jlama.JlamaLlamaTest -- First prompt 
PromptContext{prompt='<|start_header_id|>system<|end_header_id|>

Cutting Knowledge Date: December 2023
Today Date: 26 Jul 2024

<|eot_id|><|start_header_id|>user<|end_header_id|>

您能讲中文吗<|eot_id|><|start_header_id|>assistant<|end_header_id|>

', optionalTools=Optional.empty}
23:50:06.048 [main] DEBUG com.github.tjake.jlama.tensor.KvBufferCache -- Optimal page size: 28 layers, 36 context length, 8257536 bytes, 1 layer pages, 3641 length pages
23:50:06.049 [main] DEBUG com.github.tjake.jlama.model.AbstractModel -- Starting at token 0 for session 1c54efb0-8879-4604-b9f3-03ba378a6052 with prompt <|start_header_id|>system<|end_header_id|>

Cutting Knowledge Date: December 2023
Today Date: 26 Jul 2024

<|eot_id|><|start_header_id|>user<|end_header_id|>

您能讲中文吗<|eot_id|><|start_header_id|>assistant<|end_header_id|>

23:50:06.167 [main] DEBUG com.github.tjake.jlama.tensor.operations.util.JarSupport -- Loaded jlama-native library: /tmp/jlama13277383875011588717/libjlama.so
23:50:06.191 [main] DEBUG com.github.tjake.jlama.util.MachineSpec -- Machine Vector Spec: AVX_256
23:50:06.191 [main] DEBUG com.github.tjake.jlama.util.MachineSpec -- Byte Order: LITTLE_ENDIAN
23:50:06.191 [main] INFO com.github.tjake.jlama.tensor.operations.TensorOperationsProvider -- Using Native SIMD Operations (OffHeap)
23:50:08.263 [main] DEBUG com.github.tjake.jlama.model.AbstractModel -- 34 prompt tokens in 2210ms | 65.0ms per token
23:50:15.437 [main] DEBUG com.github.tjake.jlama.model.AbstractModel -- 

elapsed: 9s, prompt 65.0ms per token, gen 152.0ms per token

23:50:15.440 [main] INFO com.x.app.ai.jlama.JlamaLlamaTest -- Response: You didn't provide any information or context for me to understand what you'd like to talk about. If you have a specific question or topic in mind, feel free to share it, and I'll do my best to assist you!

please help me!

**Comment by tjake** [comment]
Hi Erik!

Which model did you use? I can take a look and let you know.  I know others have had success with this.

**Comment by zhouruicn** [comment]
Thank you for your reply. 
The model I am using is Qwen2.5-3B-Instruct, which I downloaded from Hugging Face.
https://huggingface.co/Qwen/Qwen2.5-3B-Instruct/tree/main

**Comment by tjake** [comment]
Hi,  

Yes there does seem to be an issue with the tokenizer here.  I tested some other models and `tjake/gemma-2-2b-it-JQ4` understands it as well as `tjake/Yi-Coder-1.5B-Chat-JQ4`

I'll open an issue for the Qwen and Llama

**Reply by zhouruicn** [reply]
Great job! This is exactly what I needed. I'm excited to see what comes next.

**Comment by edwardcapriolo** [comment]
The jlama tokenizer is not generally full featured. For llama models there are lots og G^ characters, etc. For some models this doent matter much for other models it is catastrophic (not identifying stop tokens etc. 

Deliverance grace is more of an exact byte for byte tested against hugging 

https://github.com/edwardcapriolo/deliverance/blob/main/core/qwen3_support.md  

I have also tested with english chinese transation

```case: chinese-roundtrip
question 1/3:
Translate this question into Chinese only: I have a square with a side length of 4. What is its area?
--------------------------------------------------------------------------------
answer 1/3:
<think>19:52:21.335 [main] INFO io.teknek.deliverance.model.AbstractModel - time_to_first_token=2549.959713 prefix_length=0

Okay, the user is asking about the area of a square with a side length of 4. Let me make sure I understand the question correctly. They want the area in Chinese, so I need to translate the question into Chinese but only the question, not the answer.

First, the original question is: "I have a square with a side length of 4. What is its area?" So, translating this into Chinese. Let's break it down.

"I have a square" would be "我有一个正方形" (Wǒ yǒu yī gè zhèngfāngxíng). "with a side length of 4" is "边长为4" (biān cháng wéi 4). Then "What is its area?" translates to "它的面积是多少？" (tā de mianjī shì méi shǎo?).

Putting it all together: "我有一个边长为4的正方形。它的面积是多少？" 

Wait, let me check if "边长" is the correct term for "side length" in this context. Yes, "边长" is commonly used for side length of a square or polygon. "正方形" is square, and "面积" is area. The structure seems correct. 

I should make sure there are no grammar issues. The sentence structure in Chinese is correct. The first part introduces the square with the given side length, and the second part asks for the area. 

Is there any alternative way to phrase it? Maybe "一个边长为4的正方形" instead of "我有一个边长为4的正方形". But the original says "I have a square", so "我有一个" is appropriate. 

Also, "side length" is "边长", and "area" is "面积". The question is straightforward. I think this translation is accurate and clear.
</think>

我有一个边长为4的正方形。它的面积是多少？
turn_result: finish=STOP_TOKEN generated=413 time_ms=74794
--------------------------------------------------------------------------------
```
